### PR TITLE
CNF-14473:nrop:hypershift: create a workflow for hcp + optional-operator

### PR DESCRIPTION
**Pull request by Tal-or**
This workflow spins up an ephemeral MNG cluster and hosted cluster
and installs a day-2 (optional) operator on the hosted cluster which
provided by the `OO_BUNDLE` variable.
    
In order to install the operator on the hosted-cluster
one should set the `USE_HOSTED_KUBECONFIG = true` under
the test step.

Signed-off-by: Talor Itzhak <titzhak@redhat.com>

### CODE REVIEW

**Comment by Tal-or** [comment]
/pj-rehearse pull-ci-openshift-kni-numaresources-operator-main-ci-e2e-install-hypershift

**Comment by Tal-or** [comment]
/pj-rehearse pull-ci-openshift-kni-numaresources-operator-main-ci-e2e-install-hypershift

**Comment by Tal-or** [comment]
/pj-rehearse pull-ci-openshift-kni-numaresources-operator-main-ci-e2e-install-hypershift

**Comment by Tal-or** [comment]
/pj-rehearse pull-ci-openshift-kni-numaresources-operator-main-ci-e2e-install-hypershift

**Comment by Tal-or** [comment]
/pj-rehearse pull-ci-openshift-kni-numaresources-operator-main-ci-e2e-install-hypershift

**Comment by Tal-or** [comment]
/pj-rehearse pull-ci-openshift-kni-numaresources-operator-main-ci-e2e-install-hypershift

**Comment by Tal-or** [comment]
/pj-rehearse pull-ci-openshift-kni-numaresources-operator-main-ci-e2e-install-hypershift

**Comment by Tal-or** [comment]
/CC @sjenning @orenc1 Hey guys, could you please review this one?

**Comment by Tal-or** [comment]
/pj-rehearse pull-ci-openshift-kni-numaresources-operator-main-ci-e2e-install-hypershift

**Comment by orenc1** [comment]
for the `optional-operators-operator-sdk` part:
/lgtm

**Comment by Tal-or** [comment]
/pj-rehearse pull-ci-openshift-kni-numaresources-operator-main-ci-e2e-install-hypershift

File: ci-operator/step-registry/hypershift/optional-operators/OWNERS [hunk 1]
```diff
@@ -0,0 +1,13 @@
+approvers:
+- csrwng
+- enxebre
+- sjenning
+- imain
```
Comment (enxebre) [inline ci-operator/step-registry/hypershift/optional-operators/OWNERS:5]: please drop imain and include Patryk-Stefanski for both reviewers approvers

Comment (Tal-or) [reply ci-operator/step-registry/hypershift/optional-operators/OWNERS]: @enxebre Done, Thanks!

**Comment by Tal-or** [comment]
/hold 
/pj-rehearse ack
@ffromani feel free to unhold if you're happy

**Pull request merged**